app/services/accept_auction.rb

```diff
@@ -19,7 +30,36 @@ def perform
     end
   end
 
-  private
+  def vendor_ineligible?
+    !winning_bidder_is_eligible_to_be_paid?
+  end
+
+  def winning_bidder_is_eligible_to_be_paid?
+    if auction_is_small_business?
+      reckoner = SamAccountReckoner.new(winning_bidder)
+      reckoner.set!
```

Comment:
this is all happening when an auction is being accepted. I think the assumption we are making is that someone cannot become sam-unverified, but they can go from small business to not small business between the time they bid and when they win.... perhaps we will want to re-visit that assumption at some point